data/transactions/logic/fields_test.go

```diff
@@ -17,12 +17,241 @@
+	dir := path.Clean(path.Dir(filename) + "/../../../protocol")
+
+	src, err := ioutil.ReadFile(path.Join(dir, "consensus.go"))
+	require.NoError(t, err)
+
+	consensusVersionNames := make([]string, 0, 64)
+	consensusVersions := make([]string, 0, 64)
+	fset := token.NewFileSet()
+	f, err := parser.ParseFile(fset, "", src, 0)
+	require.NoError(t, err)
+	for _, decl := range f.Decls {
+		if genDecl, ok := decl.(*ast.GenDecl); ok {
+			if genDecl.Tok == token.CONST {
+				if valueSpec, ok := genDecl.Specs[0].(*ast.ValueSpec); ok {
+					if strings.HasPrefix(valueSpec.Names[0].Name, "DEPRECATED") {
+						continue
+					}
+					if callExpr, ok := valueSpec.Values[0].(*ast.CallExpr); ok {
+						if literal, ok := callExpr.Args[0].(*ast.BasicLit); ok {
+							name := valueSpec.Names[0].Name
+							value := literal.Value[1 : len(literal.Value)-1]
+							consensusVersionNames = append(consensusVersionNames, name)
+							consensusVersions = append(consensusVersions, value)
+							if verbose {
+								fmt.Printf("%s - %s\n", name, value)
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	prevLogicSigVersion := ^uint64(0)
+	logicVersionByProto := make(map[uint64]protocol.ConsensusVersion, LogicSigVersion)
+	for idx, version := range consensusVersions {
+		params, ok := config.Consensus[protocol.ConsensusVersion(version)]
+		if !ok {
+			continue
+		}
+		if params.LogicSigVersion > 0 {
+			if prevLogicSigVersion == ^uint64(0) {
+				logicVersionByProto[0] = protocol.ConsensusVersion(consensusVersions[idx-1])
+			}
+			if prevLogicSigVersion != params.LogicSigVersion {
+				logicVersionByProto[params.LogicSigVersion] = protocol.ConsensusVersion(version)
+				prevLogicSigVersion = params.LogicSigVersion
+			}
+		}
+	}
+	require.Equal(t, len(logicVersionByProto), int(LogicVersion+1))
+
+	return logicVersionByProto
+}
+
+// ensure v2+ fields fail in TEAL assembler and evaluator on a version before they introduced
+// ensure v2+ fields error in v1 program
+func TestGlobalFieldsVersions(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	t.Parallel()
+	var fields []globalFieldSpec
+	for _, fs := range globalFieldSpecs {
+		if fs.version > 1 {
+			fields = append(fields, fs)
+		}
+	}
+	require.Greater(t, len(fields), 1)
+
+	logicVersionByProto := getLogicVersionProtos(t, false)
+
+	ledger := makeTestLedger(nil)
+	for _, field := range fields {
+		text := fmt.Sprintf("global %s", field.gfield.String())
+		// check assembler fails if version before introduction
+		testLine(t, text, assemblerNoVersion, "...available in version...")
+		for v := uint64(0); v < field.version; v++ {
+			testLine(t, text, v, "...available in version...")
+		}
+		testLine(t, text, field.version, "")
+
+		ops := testProg(t, text, AssemblerMaxVersion)
+
+		// check on a version before the field version
+		preLogicVersion := field.version - 1
+		preConsensusVersion := logicVersionByProto[preLogicVersion]
```

Comment:
Since it's a unit test, get the second param from the map query and require it to be true

Reply:
done